packages/csvviewer/src/widget.ts

```diff
@@ -348,6 +348,32 @@ export class CSVDocumentWidget extends DocumentWidget<CSVViewer> {
       }
     );
   }
+
+  /**
+   * Set URI fragment identifier for rows
+   */
+  setFragment(fragment: string): void {
+    let parseFragments = fragment.split('=');
+
+    // TODO: expand to allow columns and cells to be selected
+    // reference: https://tools.ietf.org/html/rfc7111#section-3
+    if (parseFragments[0] !== '#row') {
+      return;
+    }
+
+    // multiple rows, separated by semi-colons can be provided, we will just
+    // go to the top one
+    let topRow = parseFragments[1].split(';')[0];
+
+    // a range of rows can be provided, we will take the first value
+    topRow = topRow.split('-')[0];
+
+    // go to that row
+    this.context.ready.then(() => {
+      this.content.goToLine(Number(topRow));
+    });
+    this.update();
```

Comment:
I think `this.update()` probably belongs inside your `then(...)` function since that is done asynchronously.

Reply:
Thanks @afshin: I was actually wondering if it is needed at all?

Reply:
I don't think it is necessary at all, since the scrolling/viewport logic does not happen in the `onUpdateRequest` handler.